### PR TITLE
Add Versioned Component Configuration Files doc

### PR DESCRIPTION
**Pull request by mtaufen**
Adds the VCCF doc to the community archive for wg-component-standard.
While this doc predates the WG, it is a primary historical record of
ComponentConfig's early development, which was primarily driven by
that WG.

### CODE REVIEW

File: archive/wg-component-standard/component-config/README.md
Comment (justaugustus) [inline archive/wg-component-standard/component-config/README.md:7]:
```suggestion
doc by joining the `dev@kubernetes.io` mailing list
(previously `kubernetes-dev@googlegroups.com`). Please
```

File: archive/wg-component-standard/component-config/README.md
Comment (justaugustus) [inline archive/wg-component-standard/component-config/README.md:1]: Consider using the self-link

```suggestion
This copy of [Versioned Component Configuration Files](https://goo.gl/GM8KyH)
```

or

```suggestion
This copy of [Versioned Component Configuration Files][vccf-proposal]
...

[vccf-proposal]: https://goo.gl/GM8KyH
```

if you want to shorten up this line.

File: archive/wg-component-standard/component-config/README.md
Comment (justaugustus) [inline archive/wg-component-standard/component-config/README.md:20]:
```suggestion
**Last Updated:** March 28, 2018
```

to maintain emphasis

File: archive/wg-component-standard/component-config/README.md
Comment (justaugustus) [inline archive/wg-component-standard/component-config/README.md:53]:
```suggestion
        *   We _shouldn't_ ever **remove a flag** without bumping the major version of a binary. In reality we use a [flag deprecation policy](https://kubernetes.io/docs/reference/deprecation-policy/#deprecating-a-flag-or-cli) that allows us to technically break semantic versioning of the binary as long as we give advance warning.
```

to maintain emphasis

File: archive/wg-component-standard/component-config/README.md
Comment (justaugustus) [inline archive/wg-component-standard/component-config/README.md:84]:
```suggestion
**All core Kubernetes components should eventually consume their configuration via versioned configuration files, instead of command-line flags.**
```

to maintain emphasis

File: archive/wg-component-standard/component-config/README.md
Comment (justaugustus) [inline archive/wg-component-standard/component-config/README.md:91]:
```suggestion
```console
```

File: archive/wg-component-standard/component-config/README.md
Comment (justaugustus) [inline archive/wg-component-standard/component-config/README.md:144]:
```suggestion
```go
```

File: archive/wg-component-standard/component-config/README.md
Comment (justaugustus) [inline archive/wg-component-standard/component-config/README.md:197]:
```suggestion
```console
```

File: archive/wg-component-standard/component-config/README.md
Comment (justaugustus) [inline archive/wg-component-standard/component-config/README.md:208]:
```suggestion
```go
```

File: archive/wg-component-standard/component-config/README.md
Comment (justaugustus) [inline archive/wg-component-standard/component-config/README.md:284]:
```suggestion
```console
```

File: archive/wg-component-standard/component-config/README.md
Comment (justaugustus) [inline archive/wg-component-standard/component-config/README.md:300]:
```suggestion
```console
```

File: archive/wg-component-standard/component-config/README.md
Comment (justaugustus) [inline archive/wg-component-standard/component-config/README.md:315]:
```suggestion
```go
```

File: archive/wg-component-standard/component-config/README.md
Comment (justaugustus) [inline archive/wg-component-standard/component-config/README.md:400]:
```suggestion
```console
```

**Review by justaugustus — APPROVED** [review]
Thanks @mtaufen!
/hold I've made a few style suggestions, in case you want to pick them, but I'm fine w/ approving as-is
/lgtm
/approve

**Comment by mtaufen** [comment]
Whoops, I somehow missed that a review happened on this one. I'll see if I can update it in the next couple weeks. Thanks!

/remove-lifecycle stale

File: archive/wg-component-standard/component-config/README.md
Comment (mtaufen) [reply archive/wg-component-standard/component-config/README.md:1]: Seems like that short URL still works, but there's no way to manage it. Let's keep the full link since it's probably more durable.

**Comment by mtaufen** [comment]
@justaugustus updated, looks like it needs a re-lgtm, thanks for your review!

**Review by justaugustus — APPROVED** [review]
Thanks again, @mtaufen!
/lgtm
/approve

**Pull request merged**
